### DISCUSSION TITLE
Specifying static vs shared with CMake and manifest?

### DISCUSSION THREAD

**Discussion by anthonyalayo** [microsoft/vcpkg#35987] [General]
Hey there,

I'm new to Vcpkg, and I'm trying to do what I would assume is standard practice. I have Vcpkg integrated with Cmake, and my build has `BUILD_SHARED_LIBS` as an option so it can be set ON/OFF.

Vcpkg doesn't respect this global CMake var. How does one toggle between a dynamic triplet and a static triplet using a flag like this?

**Comment by Osyotr** [comment]
The correct variable is `VCPKG_TARGET_TRIPELT` https://learn.microsoft.com/en-us/vcpkg/concepts/triplets

**Reply by anthonyalayo** [reply]
That only specifies a triplet. I want to specify, dynamically, to use either shared or static libs depending if BUILD_SHARED_LIBS is set.

**Reply by autoantwort** [reply]
You can set the triplet dynamically before the `project` call

**Reply by anthonyalayo** [reply]
@autoantwort do you have any examples publicly on github I cam reference? I figured this would be standard but I can't seem to find anything.

**Reply by autoantwort** [reply]
You mean something like https://github.com/search?q=set%28VCPKG_TARGET_TRIPLET&type=code ?

**Reply by anthonyalayo** [reply]
@autoantwort thanks for the quick response. In those examples, I see that the entire triplet has to be specified. This isn't great, because I shouldn't have to mimic the triplet selection logic that is already part of vcpkg?

I build on OSX and Linux. I would like the default triplet selection logic to occur, but then simply appending `-dynamic` if `BUILD_SHARED_LIBS` is on. Otherwise, I would need to copy/paste it all just to append `-dynamic` right?

Or is there a better way?

**Reply by autoantwort** [reply]
Currently not. Is you use presets you could set the triplet there

**Reply by anthonyalayo** [reply]
@autoantwort Is there a way for presets to be automatically selected based on arch / platform/ os?

**Reply by autoantwort** [reply]
I don't know
